### DISCUSSION TITLE
Create currency field with currency selection

### DISCUSSION THREAD

**Discussion by Gotfryd** [YetiForceCompany/YetiForceCRM#16918] [How to] [answered]
Hello, I'm currently facing a problem with creating a currency field where you can select which currency will be applied.
At this moment, currency field (UiType 71) is only set for the default currency from the system and you can not change currency type.

<img width="359" alt="unchangeable" src="https://user-images.githubusercontent.com/102538667/220571529-f16eb4cb-6026-4d2d-bd07-817274149572.png">

What I want to achieve is field type with changeable currency like this (UiType 360).

![new_field](https://user-images.githubusercontent.com/102538667/220578108-ea92591e-db51-4c60-a0e1-6ff2bd2c1ede.jpg)

As you can see, I successfully changed field UiType from 71 to 360 in my MySQL database and the field works as intended but when creating new record I got error from my database about invalid format like this:

![error](https://user-images.githubusercontent.com/102538667/220579338-531f08b3-2eca-4b1a-8610-90c14c3c52a8.jpg)

Does anyone faced the similar problem and is willing to help or provide workaround / alternative solution?

**Comment by mariuszkrzaczkowski** [comment]
![github](https://user-images.githubusercontent.com/10198654/220580432-675d77b3-9bba-473d-9294-f343f1fd7399.jpg)

**Comment by rskrzypczak** [comment, marked as answer]
you also need to change the column type to `text` in database for the waluta_test field

**Reply by Gotfryd** [reply]
It worked like a charm! I don't even know how to thank you, was struggling with this for a while but now I feel such a relief. 
Thank you very much for your help in such a short time.

**Comment by Gotfryd** [comment]
(sorry for continuing the thread)

Is it also possible to create currency field with only one specific currency type? The effect I try to achieve presents like this:

<img width="357" alt="currency" src="https://user-images.githubusercontent.com/102538667/220601493-f540fb79-f5b2-4f25-b551-cb890e72415f.png">

Previous solution caused the field to be text type thereby it saves as a "weird" sequence of variables and numbers combined (which is undestandable because informations about currency type has to be saved somewhere).
I care about it returning decimal numbers instead of string.

**Reply by rskrzypczak** [reply]
the app does not have this type of field
